### ISSUE TITLE
Real line-number for errors with consolidated RUNs

### ISSUE THREAD

**Issue by alexgit2k** [hadolint/hadolint#807]
- [ ] This is a bug report
- [x] This is a feature request
- [x] I searched existing issues before opening this one

### Expected behavior

hadolint should output the real line-number of the error in the Dockerfile.

### Actual behavior

hadolint outputs only the line-number where the initial RUN is specified.

### Steps to reproduce the behavior

When consolidating RUNs like in [DL3059 ](https://github.com/hadolint/hadolint/wiki/DL3059) hadolint outputs only the line-number where the initial RUN is specified, which makes it rather difficult for long Dockerfiles when all errors appear to be in the same line.

Reproduceable with the Dockerfile below and
```
hadolint Dockerfile
```

**Output of `hadolint --version`:**

```bash
Haskell Dockerfile Linter 2.9.3-no-git
```

**Dockerfile**

```Dockerfile
FROM debian:latest
RUN apt-get update -yqq \
    && apt-get install -yqq git \
    && apt-get install -yqq vim \
    && echo "Hello World" \
    && ls -l */ \
    && apt-get clean \
    && rm -f /var/cache/apt/archives/*.deb \
    && rm -rf /var/lib/apt/lists/*
```

### Workaround

When single commands are not separated over multiple lines the following workaround can be used to get the real line-number of the error:
```
sed -i -e 's/^\s\+&&/RUN /' -e 's/\\$//' Dockerfile
hadolint --ignore DL3059 --ignore DL3009 Dockerfile
```

[DL3059 ](https://github.com/hadolint/hadolint/wiki/DL3059) (multiple consecutive RUNs) and [DL3009 ](https://github.com/hadolint/hadolint/wiki/DL3009) (apt-get-lists are not deleted on each apt-get-line) have to be ignored for the workaround.
